tls/extensions/s2n_client_status_request.c

```diff
@@ -22,29 +22,70 @@
 
 #include "utils/s2n_safety.h"
 
-int s2n_extensions_client_status_request_send(struct s2n_connection *conn, struct s2n_stuffer *out)
+static bool s2n_client_status_request_should_send(struct s2n_connection *conn);
+static int s2n_client_status_request_send(struct s2n_connection *conn, struct s2n_stuffer *out);
+static int s2n_client_status_request_recv(struct s2n_connection *conn, struct s2n_stuffer *extension);
+
+const s2n_extension_type s2n_client_status_request_extension = {
+    .iana_value = TLS_EXTENSION_STATUS_REQUEST,
+    .is_response = false,
+    .send = s2n_client_status_request_send,
+    .recv = s2n_client_status_request_recv,
+    .should_send = s2n_client_status_request_should_send,
+    .if_missing = s2n_extension_noop_if_missing,
+};
+
+static bool s2n_client_status_request_should_send(struct s2n_connection *conn)
+{
+    return conn->config->status_request_type != S2N_STATUS_REQUEST_NONE;
```

Comment:
I guess I'm a little lost on why you need to use the != operatior here. Why not use == S2N_STATUS_REQUEST_OCSP? Unless there's another value for this field that I'm missing? Or will there be more fields in the future? I'm just trying to think through what is the most readable here.

Reply:
"== S2N_STATUS_REQUEST_OCSP" would also work. My argument for the current logic would be that we don't have to update it if we needed to support a different kind of status request, but it's a pretty minor difference.